CMAKE_${_CMAKE_PROCESSING_LANGUAGE}_COMPILER_ARCHITECTURE_ID not set

Hello, and thank you for this useful tutorial.
I'm trying to follow the minimal exemple tutorial, but I got an error at the step "cmake -B_builds -G "Ninja Multi-Config" --toolchain /path/to/iar-toolchain.cmake".
In my case, I use simply "Ninja" instead of Ninja Multi-Config. Almost everything seems to be OK, but I got this error :

CMake Error at C:/Program Files/CMake/share/cmake-3.27/Modules/CMakeFindBinUtils.cmake:146 (message):
  Failed to find linker and librarian for IAR on .

I got into CMakeFindBinUtils.cmake file to understand this error message case, and I saw that the CMAKE_${_CMAKE_PROCESSING_LANGUAGE}_COMPILER_ARCHITECTURE_ID variable is empty. I think it should be "arm". Have you some elements ?

Thank you for your answer.
I tried to change the TOOLKIT variable from arm to my arm IAR path, but I still got the issue. It seems the TOOLKIT variable has no incidence at the end, the arch is empty.

Here is my iar-toolchain.cmake file.

`# Toolchain File for the IAR C/C++ Compiler

# Action: Set the `TOOLKIT` variable
# Examples: arm, riscv, rh850, rl78, rx, stm8, 430, 8051, avr or v850
# Alternative: override the default TOOLKIT_DIR (/path/to/installation/<arch>)
set(TOOLKIT "C:/Program Files/IAR Systems/Embedded Workbench 9.0/arm")

# Get the toolchain target from the TOOLKIT
get_filename_component(CMAKE_SYSTEM_PROCESSOR ${TOOLKIT} NAME)

# Set CMake for cross-compiling
set(CMAKE_SYSTEM_NAME Generic)

# IAR C Compiler
find_program(CMAKE_C_COMPILER
  NAMES icc${CMAKE_SYSTEM_PROCESSOR}
  PATHS ${TOOLKIT}
        "$ENV{ProgramFiles}/IAR Systems/*"
        "$ENV{ProgramFiles\(x86\)}/IAR Systems/*"
        /opt/iarsystems/bx${CMAKE_SYSTEM_PROCESSOR}
  PATH_SUFFIXES bin ${CMAKE_SYSTEM_PROCESSOR}/bin
  REQUIRED )

# IAR C++ Compiler
find_program(CMAKE_CXX_COMPILER
  NAMES icc${CMAKE_SYSTEM_PROCESSOR}
  PATHS ${TOOLKIT}
        "$ENV{PROGRAMFILES}/IAR Systems/*"
        "$ENV{ProgramFiles\(x86\)}/IAR Systems/*"
        /opt/iarsystems/bx${CMAKE_SYSTEM_PROCESSOR}
  PATH_SUFFIXES bin ${CMAKE_SYSTEM_PROCESSOR}/bin
  REQUIRED )

# IAR Assembler
find_program(CMAKE_ASM_COMPILER
  NAMES iasm${CMAKE_SYSTEM_PROCESSOR} a${CMAKE_SYSTEM_PROCESSOR}
  PATHS ${TOOLKIT}
        "$ENV{PROGRAMFILES}/IAR Systems/*"
        "$ENV{ProgramFiles\(x86\)}/IAR Systems/*"
        /opt/iarsystems/bx${CMAKE_SYSTEM_PROCESSOR}
  PATH_SUFFIXES bin ${CMAKE_SYSTEM_PROCESSOR}/bin
  REQUIRED )

# Avoids running the linker during try_compile()
set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)

# Set the TOOLKIT_DIR variable for the CMakeLists
get_filename_component(BIN_DIR ${CMAKE_C_COMPILER} DIRECTORY)
get_filename_component(TOOLKIT_DIR ${BIN_DIR} PATH)

unset(BIN_DIR)`

and here is my terminal output : 

`C:\DEV\zephyr_iar\cmake-tutorial\examples\arm\hello-world>cmake -B_builds -GNinja --toolchain ..\..\iar-toolchain.cmake
-- The C compiler identification is IAR
CMake Error at C:/Program Files/CMake/share/cmake-3.27/Modules/CMakeFindBinUtils.cmake:146 (message):
  Failed to find linker and librarian for IAR on .
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.27/Modules/CMakeDetermineCCompiler.cmake:196 (include)
  CMakeLists.txt:5 (project)


-- Configuring incomplete, errors occurred!

C:\DEV\zephyr_iar\cmake-tutorial\examples\arm\hello-world>`

I printed CMAKE_SYSTEM_PROCESSOR and CMAKE_C_COMPILER value to check, and they are OK : 
CMAKE_SYSTEM_PROCESSOR = arm
CMAKE_C_COMPILER = C:/Program Files/IAR Systems/Embedded Workbench 9.0/arm/bin/iccarm.exe

But after that, it seems the ARM part is missing. We can see that the difference between us is on the C compiler value seen by cmake.
You : -- The C compiler identification is IAR ARM 9.40.1
Me : -- The C compiler identification is IAR

Don't know how to fix that...

--- a/README.md
+++ b/README.md
@@ -3,7 +3,6 @@
 [CMake][url-cm-home] is an open-source, cross-platform family of tools maintained and supported by [Kitware][url-cm-kitware]. CMake is used to control the software compilation process - using simple configuration files - to generate native build scripts for a selected build system, like ninja, make, etc. For detailed documentation, visit the [CMake Documentation Page][url-cm-docs].
 
 This tutorial serves as a very basic-level guide to using CMake together with the __IAR C/C++ compilers__ to cross-compile embedded software applications for the supported target architectures.
-The core ideas presented were inspired by [Technical Note 190701][url-iar-docs-tn190701].
 
 
 ## Prerequisites
@@ -17,9 +16,11 @@ This tutorial assumes that:
 
 | __Tool__                                | __Windows-based systems__                                                                         | __Linux-based systems__                                            |
 |-----------------------------------------|---------------------------------------------------------------------------------------------------|--------------------------------------------------------------------|
-| IAR C/C++ Compiler for...               | `Arm`, `RISC-V`, `RL78`, `RX`, `RH850`,<br>`8051`, `AVR`, `MSP430`, `STM8` or `V850`              | `Arm`, `RISC-V`, `RL78`, `RX` or `RH850`                           |
+| IAR C/C++ Compiler for...               | `Arm`, `RISC-V`, `RL78`, `RX`, `RH850`,<br>`8051`, `AVR`, `MSP430`, `STM8` or `V850`              | `Arm`, `AVR`, `RISC-V`, `RL78`, `RX` or `RH850`                           |
 | [CMake 3.23+](https://cmake.org)        | Download and install the [latest][url-cmake-dl] for `-windows-x86_x64.msi`                        | Use the [Kitware APT repository](https://apt.kitware.com/) or,<br> if suitable, the distribution-provided package |
 | [Ninja 1.10+](https://ninja-build.org)  | Download the latest [ninja-win.zip][url-ninja-dl] and extract "ninja.exe" to a directory belonging to the `PATH` environment variable (like `C:\Windows\`) | Usually, the distribution-provided package should be enough |
+
+>:warning: Before being able to start using CMake, make sure the compiler license is activated. Refer to your product's documentation for details.
 
 >:penguin: Recent **Linux distributions** offer relatively up-to-date packages for `cmake` and `ninja`. Once installed, these executables are normally found on the default search path, so that both can be executed directly from anywhere in the system. You can follow the official [Kitware's APT repository](https://apt.kitware.com/) instructions so you can use it with your package manager to stay always up-to-date.
 >
@@ -216,7 +217,6 @@ This tutorial provides information on how to start building embedded software pr
 [url-repo-issue-new]:    https://github.com/IARSystems/cmake-tutorial/issues/new
 [url-repo-issue-old]:    https://github.com/IARSystems/cmake-tutorial/issues?q=is%3Aissue+is%3Aopen%7Cclosed
    
-[url-iar-docs-tn190701]: https://www.iar.com/knowledge/support/technical-notes/general/using-cmake-with-iar-embedded-workbench/
 [url-iar-docs-macros]:   https://wwwfiles.iar.com/arm/webic/doc/EWARM_DebuggingGuide.ENU.pdf#page=417
 [url-iar-docs-cspybat]:  https://wwwfiles.iar.com/arm/webic/doc/EWARM_DebuggingGuide.ENU.pdf#page=503
 [url-iar-docs-ext-elf]:  https://www.iar.com/knowledge/support/technical-notes/debugger/debugging-an-externally-built-executable-file/

